Assets/MRTK/Providers/Oculus/XRSDK/MRTK-Quest/Scripts/Config/MRTKOculusConfig.cs

```diff
@@ -137,6 +147,35 @@ public Material CustomHandMaterial
         /// </summary>
         public string PinchStrengthMaterialProperty => pinchStrengthMaterialProperty;
 
+        [Header("Teleporter Pointer Configuration")]
+        [SerializeField]
+        [Tooltip("Controls which teleport mode is utilized by MRTK-Quest controllers." +
+                 "Note to use the official pointer, you must add a parabollic pointer to your pointer input profile that supports articulated hands.")]
+        private TeleportPointerMode teleportPointerMode = TeleportPointerMode.Custom;
+
+        /// <summary>
+        /// Controls which teleport mode is utilized by MRTK-Quest controllers.
+        /// Note to use the official pointer, you must add a parabollic pointer to your pointer input profile that supports articulated hands.
+        /// </summary>
+        public TeleportPointerMode ActiveTeleportPointerMode
+        {
+#if OVRPLUGIN_UNSUPPORTED_PLATFORM
+            // If the platform is not supported by oculus, we need to ensure we don't create a teleport pointer that can't be used.
+            get => TeleportPointerMode.None;
+#else
+            get => teleportPointerMode;
+#endif
+        }
+
+        [SerializeField]
+        [Tooltip("Custom teleport pointer prefab, to be managed directly by MRTK-Quest, given that MRTK doesn't currently officially support teleport with articulated hands.")]
+        private GameObject customTeleportPointerPrefab = null;
```

Comment:
I'm not really following why the custom teleport pointer needs to be managed by MRTK-Quest? Why not just have MRTK support teleport pointers with articulated hands? What does it mean for MRTK-Quest to manage something? Isn't Quest just another platform for MRTK, not something external that needs to be added?

Reply:
Yeah reading over eric's comments on the custom teleporter pointer class I'll be looking to see if we can tear it off of it's mrtk quest dependency and make it more general to the ArticulatedHand controller type.

Reply:
@julenka the reason why the custom teleporter was managed by MRTK-Quest was because the articulated hand definition doesn't provide the ability to add extra input actions, and the parabollic pointer needs a Vector2 action to work.

Beyond that, I took the time to add extra polish, with the main improvements are just the logic around the cursor  management (making it appear only when needed and directly controlling it, rather than spawning a separate prefab), adding audio, as well as modifying some of the colors for the ray gradient, in addition to utilizing the StabilizedRay utility to smooth the the ray direction.

Also, I noticed that the parabollic pointer ray has some dot product check that checks the ray's alignment to the ground, and scales the ray velocity accordingly. I modified that math to make it move more responsively when youre teleporting near yourself.

Reply:
>doesn't provide the ability to add extra input actions, and the parabollic pointer needs a Vector2 action to work.

Could you fire an arbitrary input event based on a teleport input action mapped in a device manager profile? For example, have a

```
[SerializeField]
private MixedRealityInputAction teleportAction;

public MixedRealityInputAction TeleportAction => teleportAction;
```

It seems like there's already code to handle this case:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/4e60a8bcbde64dd05fd86062b314736b6d7d629f/Assets/MRTK/Providers/Oculus/XRSDK/MRTK-Quest/Scripts/Input/Controllers/OculusHand.cs#L336-L339